rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.vectors/180_update_dense_vector_type.yml

```diff
@@ -0,0 +1,1401 @@
+---
+"Test create and update dense vector mapping with per-doc indexing and flush":
+  - skip:
+      version: " - 8.13.99"
+      reason: "introduced after 8.13"
+  - do:
+      indices.create:
+        index: test_index
+
+  - do:
+      indices.put_mapping:
+        index: test_index
+        body:
+          properties:
+            embedding:
+              type: dense_vector
+              dims: 4
+              index_options:
+                type: flat
+
+  - do:
+      indices.get_mapping:
+        index: test_index
+
+  - match: { test_index.mappings.properties.embedding.type: dense_vector }
+  - match: { test_index.mappings.properties.embedding.index_options.type: flat }
+
+  - do:
```

Comment:
❓  Is it worth checking both _update and _bulk? What could be the difference in terms of indexing? 🤔

Reply:
it's mostly to be on the safe side, as they would result in different set of segments being created (e.g. and different HNSW graphs too). The key thing here is making sure indexing and search keep working well even when doing this mapping update, so I'm trying to cover as many cases as possible.

Reply:
super-newbie Lucene question: If we're not doing `refresh` after each `index` request - but just a `refresh` at the end, would that be different than doing a `bulk` with a `refresh` at the end? (supposing we don't hit the `refresh_interval` before finishing indexing)

Reply:
the idea is to make it _possible_ to have Lucene merges happening before the refresh, which would result in possibly different underlying segments being created, Lucene-wise the two setups (_bulk_ vs _index_) should result in different commits too.